test/index.spec.js

```diff
@@ -10,10 +10,52 @@ describe("Testing Rewrite", () => {
     expect(input).to.deep.equal({});
   });
 
-  it("Testing Retain Empty List", () => {
-    const input = { test: [] };
-    index({ retain: ["test"] })(input);
-    expect(input).to.deep.equal({ test: [] });
+  describe("Testing Retain Empty List", () => {
+    const execTest = (retain, retainEmptyParents) => {
+      const input = { test: [] };
+      index({ retain, retainEmptyParents })(input);
+      return input;
+    };
+
+    it("Testing Retain Empty List (Exact Match), retainEmptyParents = true", () => {
+      expect(execTest(["test"], true)).to.deep.equal({ test: [] });
+    });
+
+    it("Testing Retain Empty List (Exact Match), retainEmptyParents = false", () => {
+      expect(execTest(["test"], false)).to.deep.equal({ test: [] });
+    });
+
+    it("Testing Retain Empty List (Child Match), retainEmptyParents = true", () => {
+      expect(execTest(["test.id"], true)).to.deep.equal({ test: [] });
+    });
+
+    it("Testing Retain Empty List (Child Match), retainEmptyParents = false", () => {
+      expect(execTest(["test.id"], false)).to.deep.equal({});
+    });
+
+    it("Testing Retain Empty List (No Match), retainEmptyParents = true", () => {
+      expect(execTest(["unknown"], true)).to.deep.equal({});
+    });
+
+    it("Testing Retain Empty List (No Match), retainEmptyParents = false", () => {
```

Comment:
Pretty verbose 'it' names that match the parent 'describe', maybe unnecessary?

Reply:
The problem is the naming collision complains... Not sure what best practises are here? Any idea?

Reply:
If 2 `it`s have the same name across multiple `describe`s they have naming collisions? That seems like goofy behaviour. Not a lot you can do about that though...

Reply:
Yeah, that's unfortunately how it would seem...